Source/ARTDefaultLocalDeviceStorage.m

```diff
@@ -0,0 +1,23 @@
+#import "ARTLog.h"
+#import "ARTLocalDevice+Private.h"
+#import "ARTDefaultLocalDeviceStorage.h"
+
+@implementation ARTDefaultLocalDeviceStorage
```

Comment:
Fixed efed0c1

>what are the security implications of storing this secret in the user defaults instead of as a keychain

We definitely should tell users both in the library and the docs how it will affect the security of their apps. See also comment below.

Reply:
> What are the consequences of a re-registration? Would we like to avoid it, or does it not matter? We could avoid it by trying to migrate the existing secret from the keychain, if we wanted to.

Did you have any thoughts about this bit?

Reply:
@ben-xD Can you assist here? Does re-registration trigger re-login? Or how does it look like from the user's POV?

Reply:
I don't know what you mean by Re-login? What do you mean from the user perspective? @maratal.

Reply:
> We definitely should tell users both in the library and the docs how it will affect the security of their apps. See also comment below.

Yes definitely, it should be as part of this PR IMHO :).

Reply:
Anyway, if we suddenly change storage mechanism, if a client (device running an App using Ably Cocoa) updates with this new release Ably Cocoa, they will lose access to their device registration. 

That is very bad. Push notifications may still be received as the APNs device token does not change, but there is no way to update this device registration or delete it, since the client has lost track of it. 

An example, a device is activated for push notifications, and has a device registration on Ably. If a user logs out, an app developer might decide to deactivate the device, or delete the device registration in Ably. In this case, we have lost the deviceId so they won't be able to delete it. Therefore, a new user logging into the app will receive notifications from the previous user, since the APNs device token is the same.

Not only that, we would leave a lot of old outdated registrations untouched in the database when we have to register a new one.

We should have code to migrate data from keychain to UserDefaults.

Reply:
>Should we have code to migrate data from keychain to UserDefaults?

@ben-xD The last commit does exactly that.

Reply:
Okay, cool: efed0c1cdbd12af99069b3118d87b0cce86aa0fc